db/migrate/20150116095004_patch_corrupt_attachments.rb

```diff
@@ -0,0 +1,49 @@
+##
+# Goes through all attachments looking for ones whose 'file' column, which is the new column
+# used by the carrierwave-based attachments, is not set.
+#
+# For every one of those attachments the migration then sets the 'file' column to
+# whatever the value of the legacy column 'filename' is. If that one is empty too
+# it falls back to the 'disk_filename' column. This one was not meant to be displayed
+# to users but it's better than nothing, especially when trying to identify corrupt attachments.
+#
+# If *that* column is empty too, the attachment is broken beyond repair and will be dropped.
+#
+# Note: Just because the 'file' column is restored doesn't mean the actual file exists.
+#       Rather the 'file' column being empty means precisely that the file is missing.
+#       By still writing the filename into the file column the attachment can at least
+#       be displayed, if not downloaded.
+#
+# Important: The migration is irreversible.
+class PatchCorruptAttachments < ActiveRecord::Migration
+  def up
+    Attachment.all.each do |attachment|
+      patch_attachment attachment
+    end
+  end
+
+  def down
+    puts "Can't revert this migration as it would mean breaking valid attachments. \
+          Even if we wanted to do that we can't know which ones were broken before \
+          to only break those again.".squish
```

Comment:
Table before the migration (after the first one that skipped corrupt attachments):

| filename | diskfilename | file|
|-------------|------------------|------|
| found.jpg | 20131107_found.jpg | found.jpg |
| not_found.gif | 20140503_not_found.gif | |

And after this migration:

| filename | diskfilename | file|
|-------------|------------------|------|
| found.jpg | 20131107_found.jpg | found.jpg |
| not_found.gif | 20140503_not_found.gif | not_found.gif |

So from the data in the table alone we can not deduct whether or not the attachment was broken before.
We could, however, check if the file can be found and if not we just delete the file column and we would have
the same state as before. But yes, that would corrupt things even more. I can adjust the error message with this bit.